### ISSUE TITLE
nae_delta module fails with: AttributeError: 'module' object has no attribute 'decompress'

### ISSUE THREAD

**Issue by 020003** [CiscoDevNet/ansible-nae#38]
Hi,

today I tried to do a simple test with the nae_delta module. 
However I run in an issue on multiple ansible hosts with different versions.
Versions tested: 2.9.3, 2.9.12

`fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1606990806.61-7624-224921613645382/AnsiballZ_nae_delta.py\", line 102, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1606990806.61-7624-224921613645382/AnsiballZ_nae_delta.py\", line 94, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1606990806.61-7624-224921613645382/AnsiballZ_nae_delta.py\", line 40, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.cisco.nae.plugins.modules.nae_delta', init_globals=None, run_name='__main__', alter_sys=True)\n  File \"/usr/lib64/python2.7/runpy.py\", line 176, in run_module\n    fname, loader, pkg_name)\n  File \"/usr/lib64/python2.7/runpy.py\", line 82, in _run_module_code\n    mod_name, mod_fname, mod_loader, pkg_name)\n  File \"/usr/lib64/python2.7/runpy.py\", line 72, in _run_code\n    exec code in run_globals\n  File \"/tmp/ansible_cisco.nae.nae_delta_payload_f12p0D/ansible_cisco.nae.nae_delta_payload.zip/ansible_collections/cisco/nae/plugins/modules/nae_delta.py\", line 121, in <module>\n  File \"/tmp/ansible_cisco.nae.nae_delta_payload_f12p0D/ansible_cisco.nae.nae_delta_payload.zip/ansible_collections/cisco/nae/plugins/modules/nae_delta.py\", line 102, in main\n  File \"/tmp/ansible_cisco.nae.nae_delta_payload_f12p0D/ansible_cisco.nae.nae_delta_payload.zip/ansible_collections/cisco/nae/plugins/module_utils/nae.py\", line 1751, in new_delta_analysis\n  File \"/tmp/ansible_cisco.nae.nae_delta_payload_f12p0D/ansible_cisco.nae.nae_delta_payload.zip/ansible_collections/cisco/nae/plugins/module_utils/nae.py\", line 961, in get_epochs\nAttributeError: 'module' object has no attribute 'decompress'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
`

The playbook looks like this: 

`- name: Play 1 Check Epoch Delta
  hosts: all
  connection: local

  tasks:

   - name: Create delta analysis from two most recent epochs
     cisco.nae.nae_delta:
       host: "{{ nae }}"
       port: 443
       username: "{{ lookup('env', 'ANSIBLE_NET_USERNAME') }}"
       password: "{{ lookup('env', 'ANSIBLE_NET_PASSWORD') }}"
       ag_name: "{{ agName }}"
       name: Delta_Analysis_1
       state: present

`

Please Advice :)

**Comment by camrossi** [comment]
Hi,
The issue is that gzip does not have the decompress method in Python2.7

See the error at line [961](https://github.com/CiscoDevNet/ansible-nae/blob/master/plugins/module_utils/nae.py#L961) 
`'module' object has no attribute 'decompress'\n" 
`
If you see there we do gzip.decompress. 
gzip.decompress is only available with Python3 but you are using Python 2.7 

We do not support pyton 2.7 as is End Of Line and End Of Support as if January 1, 2020.
See [This](https://www.python.org/doc/sunset-python-2)

Bottom line you need to upgrade to Python 3 :)

**Comment by 020003** [comment]
Yep you are right somehow it was still trying to use the python 2.7 interpreter... even if it was executed in a python 3 venv....
Thx for the quick response ;)
